src_python/habitat_sim/utils/settings.py

```diff
@@ -8,27 +8,26 @@
 default_sim_settings = {
     "scene_dataset_config_file": "default",
     "scene": "NONE",
-    "width": 640,
-    "height": 480,
     "default_agent": 0,
-    "sensor_height": 1.5,
-    "hfov": 90,
-    "color_sensor": True,
-    "semantic_sensor": False,
-    "depth_sensor": False,
-    "ortho_rgba_sensor": False,
-    "ortho_depth_sensor": False,
-    "ortho_semantic_sensor": False,
-    "fisheye_rgba_sensor": False,
-    "fisheye_depth_sensor": False,
-    "fisheye_semantic_sensor": False,
-    "equirect_rgba_sensor": False,
-    "equirect_depth_sensor": False,
-    "equirect_semantic_sensor": False,
     "seed": 1,
     "physics_config_file": "data/default.physics_config.json",
+    "sensors": [
+        {
+            "uuid": "color_sensor",
+            "width": 640,
+            "height": 480,
+            "sensor_height": 1.5,
+            "hfov": 90,
+            "sensor_subtype": habitat_sim.SensorSubType.PINHOLE,
+            "sensor_type": habitat_sim.SensorType.COLOR,
+            "position": [0, 1.5, 0],
+            "channels": 3,
+            "camera_type": habitat_sim.CameraSensorSpec,
+        },
```

Comment:
I suggest we pull this out as a default_sensor_settings dict which is copied into the default settings dict.

Also, let's simplify a bit and remove "sensor_subtype", "sensor_type", "channels", "camera_type", since those are determined by the sensor type and users shouldn't need to both with these.

"sensor_height" is redundant if we have "position", so let's remove that one.

Reply:
sensor settings have been pulled out in its own dictionary. Sensor type, sensor subtypes etc are inferred from uuids, similar to how the original make_cfg fn did so, so fn won't work perfectly unless uuid is well-labeled, but I think that's a reasonable tradeoff.